### PR TITLE
feat: vendors Zarf inside of UDS CLI

### LINKED ISSUE #167: Vendor Zarf inside of UDS CLI
### Describe what should be investigated or refactored
Today, to deploy the Zarf init package (and perform certain Zarf actions), the Zarf binary is a mandatory dependency of UDS CLI
- Example: The `init` package attempts to create Gitea users using `internal` Zarf commands that don't exist in UDS CLI. If a user attempts to install the Zarf `init` package using UDS CLI but not the Zarf binary, the `init` package will install without a fatal error but the GItea users won't be created

### Additional context
- Related issue: https://github.com/defenseunicorns/uds-cli/issues/158

### PR DESCRIPTION
**Pull request by UncleGedd**
## Description
Vendors Zarf inside of UDS CLI to remove UDS's dependency on a system Zarf

## Related Issue

Fixes #167 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md)(https://github.com/defenseunicorns/uds-template-capability/blob/main/CONTRIBUTING.md#submitting-a-pull-request) followed

### CODE REVIEW

File: .github/workflows/test-e2e.yaml [hunk 1]
```diff
@@ -32,8 +32,8 @@ jobs:
       - name: Setup golang
         uses: ./.github/actions/golang
 
-      - name: Install Zarf
-        uses: ./.github/actions/zarf
+#      - name: Install Zarf
```
Comment (decleaver) [inline .github/workflows/test-e2e.yaml:35]: kill it

**Review by decleaver — COMMENTED** [review]
pulled it down and played with it, works like a charm! besides the minor clean up, looks good to me!!

File: .github/workflows/test-e2e.yaml (hunk 1 above)
Comment (UncleGedd) [reply .github/workflows/test-e2e.yaml]: killed it

**Review by decleaver — APPROVED** [review]
ship it

**Pull request merged**